internal/controllers/cluster/cluster_controller_phases.go

```diff
@@ -218,6 +218,8 @@ func (r *Reconciler) reconcileInfrastructure(ctx context.Context, cluster *clust
 
 // reconcileControlPlane reconciles the Spec.ControlPlaneRef object on a Cluster.
 func (r *Reconciler) reconcileControlPlane(ctx context.Context, cluster *clusterv1.Cluster) (ctrl.Result, error) {
```

Comment:
So this means that the first reconcile that sets Cluster.spec.controlPlaneEndpoint will always return (and then log) an error?

This seems pretty strange behavior. Especially considering that the not ready case is neither a requeue nor error (requeue is not needed because we get a reconcile enqueued if the referenced object is updated)

Makes me wonder if we should have a separate reconcileControlPlaneEndpoint which only returns an error if the endpoint is set on none of the objects

Reply:
Note that in the last push, I added a check to the control plane to not error when `util.ErrUnstructuredFieldNotFound` is returned

Reply:
Maybe we can do the same for infrastructure ref?

Reply:
> wdym by this?

Basically the following:
* control plane endpoint becomes available on the CP
* reconcile
  * reconcileInfrastructure => returns error because endpoint is not on the infra cluster
  * reconcileControlPlane => works
  * reconcile func returns an error to controller-runtime which is then logged
  * Basically this reconcile failed even though everythign is fine. Of course the next reconcile will just work

Reply:
> Note that in the last push, I added a check to the control plane to not error when util.ErrUnstructuredFieldNotFound is returned
> Maybe we can do the same for infrastructure ref?

Yup. Let's do that. I think that's fair. Basically it is not an error just because the controlPlaneEndpoint is not set yet, just business as usual